### PR TITLE
[TASK] Payments: Use `counterparty` in place of `issuer`

### PR DESCRIPTION
**Pull request by lumberj**
- This completes the conversion of all APIs in REST from using `issuer`
  to `counterparty`

### CODE REVIEW

File: api/lib/rest-to-tx-converter.js [hunk 1]
```diff
@@ -18,12 +18,32 @@ function RestToTxConverter() {};
  */
 RestToTxConverter.prototype.convert = function(payment, callback) {
   try {
+
+
+    // TODO
```
Comment (lumberj) [inline api/lib/rest-to-tx-converter.js]: TODO to make this a non-breaking change

File: README.md [hunk 2]
```diff
@@ -869,7 +869,7 @@ You can then choose one of the returned payment objects, modify it as desired (f
       "destination_amount": {
         "value": "1",
         "currency": "USD",
-        "issuer": "rMwjYedjc7qqtKYVLiAccJSmCwih4LnE2q"
+        "counterparty": "rMwjYedjc7qqtKYVLiAccJSmCwih4LnE2q"
       },
       "invoice_id": "",
       "paths": "[[{\"account\":\"rvYAfWj5gh67oV6fW32ZzP3Aw4Eubs59B\",\"type\":1,\"type_hex\":\"0000000000000001\"},{\"currency\":\"USD\",\"issuer\":\"rMwjYedjc7qqtKYVLiAccJSmCwih4LnE2q\",\"type\":48,\"type_hex\":\"0000000000000030\"},{\"account\":\"rMwjYedjc7qqtKYVLiAccJSmCwih4LnE2q\",\"type\":1,\"type_hex\":\"0000000000000001\"}],[{\"account\":\"rvYAfWj5gh67oV6fW32ZzP3Aw4Eubs59B\",\"type\":1,\"type_hex\":\"0000000000000001\"},{\"currency\":\"XRP\",\"type\":16,\"type_hex\":\"0000000000000010\"},{\"currency\":\"USD\",\"issuer\":\"rMwjYedjc7qqtKYVLiAccJSmCwih4LnE2q\",\"type\":48,\"type_hex\":\"0000000000000030\"},{\"account\":\"rMwjYedjc7qqtKYVLiAccJSmCwih4LnE2q\",\"type\":1,\"type_hex\":\"0000000000000001\"}]]",
```
Comment (lumberj) [inline README.md]: Do we want to keep this behavior of returning the `paths` object as a string?

File: api/lib/utils.js [hunk 3]
```diff
@@ -59,18 +60,30 @@ function parseLedger(ledger) {
...
-      counterparty: currencyAmount.issuer,
-      value: currencyAmount.value
+      currency: rippledAmount.currency,
+      counterparty: rippledAmount.issuer,
+      value: rippledAmount.value
+    };
+  }
+}
+
+function txFromRestAmount(restAmount) {
```
Comment (lumberj) [inline api/lib/utils.js]: TODO: Need to write unit tests for this

**Comment by lumberj** [comment]
Currently, this is a non-backwards compatible change. I'm looking into making this backwards compatible

**Comment by lumberj** [comment]
Not-LGTM. Found a bug here, https://github.com/ripple/ripple-rest/pull/292/files?diff=unified#diff-211ab5f387c8a064258cd43d111e665dR80

where paths can be specified as a serialized json string or object. The object is the thing that REST returns, which will use `counterparty`, and the string is using `issuer` because that's how it's encoded in `payment` objects in rest (https://ripple.com/build/ripple-rest/#payment-objects)

I need to write a few more tests to cover these cases

**Comment by lumberj** [comment]
Looking further, I don't see a case where `paths` is an object and not a `string` in the `payment` object sent to Ripple-REST. REST always returns `paths` as a string

**Comment by lumberj** [comment]
No more changes to this PR

**Comment by geertweening** [comment]
LGTM

**Pull request merged**